lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -404,61 +405,95 @@ impl<'s> SegmentHolder {
             .collect()
     }
 
-    /// Select all point occurrences in all segments, regardless of their version.
-    fn all_segments_points(&self, ids: &[PointIdType]) -> AHashMap<SegmentId, Vec<PointIdType>> {
-        self.iter()
-            .map(|(segment_id, segment)| {
-                let segment_arc = segment.get();
-                let segment_lock = segment_arc.read();
-                let segment_points = Self::segment_points(ids, segment_lock.deref());
-                (*segment_id, segment_points)
-            })
-            .collect()
-    }
-
-    /// Select what point IDs are in what segments.
+    /// Select what point IDs to update and delete in each segment
+    ///
+    /// Each external point ID might have multiple point versions across all segments.
     ///
-    /// For each external id, picks the segment which has the latest version of such point.
-    fn latest_segments_points(&self, ids: &[PointIdType]) -> AHashMap<SegmentId, Vec<PointIdType>> {
-        // Find in which segments latest point versions are located
-        let mut points: AHashMap<PointIdType, (SeqNumberType, SegmentId)> =
+    /// This finds all point versions and groups them per segment. The newest point versions are
+    /// selected to be updated, all older versions are marked to be deleted.
+    ///
+    /// Points that are already soft deleted are not included.
+    fn find_points_to_update_and_delete(
```

Comment:
Function is adjusted to now collect both the latest point versions, and all older point versions.